src/compiler/scala/tools/nsc/ast/Trees.scala

```diff
@@ -174,8 +174,7 @@ trait Trees extends scala.reflect.internal.Trees { self: Global =>
   @nowarn("""cat=deprecation&origin=scala\.tools\.nsc\.ast\.Trees\.Transformer""")
   final type AstTransformer = Transformer
 
-  // TODO: deprecate when we can cleanly cross-compile without warnings
-  // @deprecated("use AstTransformer instead", since = "2.13.4")
+  @deprecated("use AstTransformer instead", since = "2.13.5")
```

Comment:
This and `Cursor` were requests from Seb to keep undeprecated because compiler plugins rely on them heavily, and we want to preserve simple cross-compilation. I *think* that means we can deprecate it when 2.12 is no longer supported, though I'm not sure if we need to wait for 2.13 not to be supported, since earlier releases of 2.13 have the same issue 2.12 has, and compiler plugins are built against the full Scala version.

Reply:
aha — I misunderstood the comment.

Reply:
Yes, it does need commenting. I made these "class shadowing" edits a long time ago on some long-forgotten branch. I don't see anything wrong with deprecation. It's deprecated instead of removed outright.